qiskit/algorithms/evolvers/variational/variational_principles/imaginary_mc_lachlan_variational_principle.py

```diff
@@ -0,0 +1,81 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class for an Imaginary McLachlan's Variational Principle."""
+
+from typing import Union, List
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import StateFn, OperatorBase, ListOp
+from ..calculators import (
+    metric_tensor_calculator,
+    evolution_grad_calculator,
+)
+from .imaginary_variational_principle import (
+    ImaginaryVariationalPrinciple,
+)
+
+
+class ImaginaryMcLachlanVariationalPrinciple(ImaginaryVariationalPrinciple):
+    """Class for an Imaginary McLachlan's Variational Principle. It aims to minimize the distance
+    between both sides of the Wick-rotated Schrödinger equation with a quantum state given as a
+    parametrized trial state. The principle leads to a system of linear equations handled by the
+    `~qiskit.algorithms.evolvers.variational.solvers.VarQTELinearSolver` class. The imaginary
+    variant means that we consider imaginary time dynamics.
+    """
+
+    def calc_metric_tensor(
+        self,
+        ansatz: Union[StateFn, QuantumCircuit],
+        parameters: List[Parameter],
+    ) -> ListOp:
+        """
+        Calculates a metric tensor according to the rules of this variational principle.
+
+        Args:
+            ansatz: Quantum state in the form of a parametrized quantum circuit to be used for
+                calculating a metric tensor.
+            parameters: Parameters with respect to which gradients should be computed.
+
+        Returns:
+            Transformed metric tensor.
+        """
+        metric_tensor_real = metric_tensor_calculator.calculate(
+            ansatz, parameters, self._qfi_method
+        )
+
+        return metric_tensor_real * 0.25
```

Comment:
See comments above about the factor 4 of the Quantum Geometric Tensor vs. Quantum Fisher Info